prometheus/registry.go

```diff
@@ -260,20 +287,27 @@ func (errs MultiError) MaybeUnwrap() error {
 type Registry struct {
 	mtx                   sync.RWMutex
 	collectorsByID        map[uint64]Collector // ID is a hash of the descIDs.
+	collectorsByCompatID  map[uint64]Collector
 	descIDs               map[uint64]struct{}
+	compatDescIDs         map[uint64]struct{}
 	dimHashesByName       map[string]uint64
+	compatDimHashesByName map[string]uint64
 	uncheckedCollectors   []Collector
 	pedanticChecksEnabled bool
+	utf8Collision         bool
```

Comment:
So we're adding a uint64, empty struct, and Collector interface obj for every metric registered.  This could be a lot of memory with many millions of metrics.  Perhaps we could only store the compat IDs if they are *different* from the default?

Reply:
(I have now implemented this)